plugins/modules/ndo_l3out_bgp_peer.py

```diff
@@ -0,0 +1,800 @@
+    export_route_map_identifier = get_object_identifier(module.params.get("export_route_map_uuid"), module.params.get("export_route_map"))
+    peer_prefix_identifier = get_object_identifier(module.params.get("peer_prefix_uuid"), module.params.get("peer_prefix"))
+
+    bgp_controls = module.params.get("bgp_controls")
+    peer_controls = module.params.get("peer_controls")
+    address_families = module.params.get("address_families")
+    private_as_controls = module.params.get("private_as_controls")
+    state = module.params.get("state")
+
+    mso_template = MSOTemplate(mso, "l3out", template_identifier.get("name"), template_identifier.get("id"))
+    mso_template.validate_template("l3out")
+
+    tenant_id = mso_template.template_summary.get("tenantId")
+    tenant_name = mso_template.template_summary.get("tenantName")
+
+    l3out_object = mso_template.get_l3out_object(uuid=l3out_identifier.get("uuid"), name=l3out_identifier.get("name"), fail_module=True)
+    node_group_object = mso_template.get_l3out_node_group(node_group, l3out_object.details, fail_module=True)
+
+    bgp_peer_objects = get_bgp_peer_by_address(mso_template, node_group_object.details.get("bgpPeers", []), ipv4_addr=ipv4_addr, ipv6_addr=ipv6_addr)
+
+    if bgp_peer_objects and (ipv4_addr or ipv6_addr):
+        set_bgp_peer_relations_name(mso, bgp_peer_objects.details)
+        mso.existing = mso.previous = copy.deepcopy(bgp_peer_objects.details)  # Query a specific object
+    elif bgp_peer_objects:
+        mso.existing = [set_bgp_peer_relations_name(mso, bgp_peer) for bgp_peer in bgp_peer_objects]  # Query all objects
+
+    if state != "query":
+        bgp_peer_path = "/l3outTemplate/l3outs/{0}/nodeGroups/{1}/bgpPeers/{2}".format(
+            l3out_object.index, node_group_object.index, bgp_peer_objects.index if bgp_peer_objects else "-"
+        )
+
+    ops = []
+
+    if state == "present":
+        peer_prefix_uuid = None
+
+        if peer_prefix_identifier.get("uuid") is not None:
+            peer_prefix_uuid = peer_prefix_identifier.get("uuid")
+        elif peer_prefix_identifier.get("name") is not None and peer_prefix_identifier.get("name") != "":
+            peer_prefix_uuid = mso_template.get_object_by_key_value_pairs(
+                "BGP Peer Prefix Policy",
+                mso.query_objs(generate_api_endpoint("templates/objects", **{"type": "bgpPeerPrefixPol", "tenant-id": tenant_id})),
+                [KVPair("name", peer_prefix_identifier.get("name"))],
+                True,
+            ).details.get("uuid")
+
+        route_map_objects = (
+            mso.query_objs(generate_api_endpoint("templates/objects", **{"type": "routeMap", "tenant-id": tenant_id}))
+            if import_route_map_identifier.get("name") or export_route_map_identifier.get("name")
+            else []
+        )
+
+        import_route_map_uuid = None
+        export_route_map_uuid = None
```

Comment:
I am not sure why this is done here, why not do it at the beginning where we do module.params.get? Like first you create a dict, but then later you are still can declare the individual variables? Is it not easier to write the uuid to the variable module.params.get and then when this value is None set this variable with mso_template.get_route_map() method?